matsim/src/main/java/org/matsim/analysis/PHbyModeCalculator.java

```diff
@@ -72,118 +69,116 @@ void addIteration(int iteration, IdMap<Person, Plan> map) {
-                        .mapToDouble(k->k.getOrDefault(mode,new TravelTimeAndWaitTime(0.0, 0.0)).waitTime/3600.0)
-                        .toArray();
-                chart.addSeries(mode + WAIT_TIME_SUFFIX, valueWaitTime);
-            }
-            chart.addMatsimLogo();
-            chart.saveAsPng(controllerIO.getOutputFilename(FILENAME+ ".png"), 1024, 768);
-
-        }
-
-    }
-
-    private static class TravelTimeAndWaitTime {
-    	private double travelTime;
-    	private double waitTime;
+	}
+
+	private void writeValues(CSVPrinter csvPrinter, TreeSet<String> allModes) throws IOException {
+		for (Map.Entry<Integer,Map<String,TravelTimeAndWaitTime>> e : phtPerIteration.entrySet()){
+			csvPrinter.print(e.getKey());
+			for (String mode : allModes){
+				TravelTimeAndWaitTime travelTimeAndWaitTime = e.getValue().getOrDefault(mode, new TravelTimeAndWaitTime(0.0, 0.0));
+				csvPrinter.print((int) Math.round(travelTimeAndWaitTime.travelTime / 3600.0));
+				csvPrinter.print((int) Math.round(travelTimeAndWaitTime.waitTime / 3600.0));
+			}
+			csvPrinter.println();
+		}
+	}
 
-    	private TravelTimeAndWaitTime(double travelTime, double waitTime) {
-    		this.travelTime = travelTime;
-    		this.waitTime = waitTime;
-    	}
+	private static void writeHeader(CSVPrinter csvPrinter, TreeSet<String> allModes) throws IOException {
+		csvPrinter.print("Iteration");
+		for (String mode: allModes) {
+			csvPrinter.print(mode + TRAVEL_TIME_SUFFIX);
+			csvPrinter.print(mode + WAIT_TIME_SUFFIX);
+		}
+		csvPrinter.println();
+	}
+
+	private void writePng(){
+		TreeSet<String> allModes = getAllModes();
+		String[] categories = new String[phtPerIteration.size()];
+		int i = 0;
+		for (Integer it : phtPerIteration.keySet()){
+			categories[i++] = it.toString();
+		}
 
-    	private static TravelTimeAndWaitTime sum(TravelTimeAndWaitTime object1, TravelTimeAndWaitTime object2) {
+		StackedBarChart chart = new StackedBarChart("Passenger hours traveled per Mode","Iteration","person hours",categories);
+		//rotate x-axis by 90degrees
+		chart.getChart().getCategoryPlot().getDomainAxis().setCategoryLabelPositions(CategoryLabelPositions.UP_90);
+
+		for (String mode : allModes){
+			double[] valueTravelTime =  phtPerIteration.values().stream()
+													   .mapToDouble(k->k.getOrDefault(mode,new TravelTimeAndWaitTime(0.0, 0.0)).travelTime/3600.0)
+													   .toArray();
+			chart.addSeries(mode + TRAVEL_TIME_SUFFIX, valueTravelTime);
+			double[] valueWaitTime =  phtPerIteration.values().stream()
+													 .mapToDouble(k->k.getOrDefault(mode,new TravelTimeAndWaitTime(0.0, 0.0)).waitTime/3600.0)
+													 .toArray();
+			chart.addSeries(mode + WAIT_TIME_SUFFIX, valueWaitTime);
+		}
+		chart.addMatsimLogo();
+		chart.saveAsPng(controllerIO.getOutputFilename(FILENAME+ ".png"), 1024, 768);
+	}
+
+	private TreeSet<String> getAllModes() {
+		return this.phtPerIteration.values()
+								   .stream()
```

Comment:
the indentation looks a bit odd, at least here on GitHub

Reply:
You are right. For some reason IntelliJ used tabs instead of spaces for line indent. Thanks for the hint!

Reply:
@paulheinr please have in mind that tabs are preferred, while spaces are not (in MATSim). See https://matsim.org/docs/devguide/conventions 
So if you fix it, please stay with tabs ;)

Reply:
Okay, that's interesting. Then I will stay with tabs. But they seem a bit outdated to me.

At the code sprint we were talking about introducing the google code formatter. They also use spaces. But this is maybe not the right place for this discussion.